src/c_kzg_4844.c

```diff
@@ -1766,39 +1799,76 @@ C_KZG_RET LOAD_TRUSTED_SETUP(
  * @param[out] out Pointer to the loaded trusted setup data
  * @param[in]  in  File handle for input
  */
```

Comment:
I would suggest adding a comment with regards to seeking operations.
Notably: Either clarify that the function expects that the file's read pointer is at the start  or actually perform fseek. 
Similar, the post-state with regards to position in the filed stream should be documented.

(Notably, calling this function twice on the same openend file will fail, which might be unexpected)

Reply:
Another good point. Alternatively, we could fseek to the beginning of the file before & after reading it.

Reply:
I would prefer to keep the logic as-is, just document it. The reason being that some FILE objects do not support arbitrary fseeks and this adds more potential failure cases. The caller is better equipped to deal with such cases.